CI latest ipython broke typing

```

mypy.....................................................................Failed
- hook id: mypy
- duration: 180.87s
- exit code: 1

pandas/io/formats/printing.py:264: error: Incompatible types in assignment (expression has type "str", base class "BaseFormatter" defined the type as "ObjectName")  [assignment]
pandas/io/formats/printing.py:265: error: Incompatible types in assignment (expression has type "Tuple[Type[Dict[Any, Any]]]", base class "BaseFormatter" defined the type as "Type[str]")  [assignment]
pandas/tests/test_register_accessor.py:16: note: By default the bodies of untyped functions are not checked, consider using --check-untyped-defs  [annotation-unchecked]
pandas/tests/io/xml/test_xml.py:1118: note: By default the bodies of untyped functions are not checked, consider using --check-untyped-defs  [annotation-unchecked]
pandas/tests/io/xml/test_xml.py:1297: note: By default the bodies of untyped functions are not checked, consider using --check-untyped-defs  [annotation-unchecked]
pandas/tests/io/xml/test_to_xml.py:995: note: By default the bodies of untyped functions are not checked, consider using --check-untyped-defs  [annotation-unchecked]
Found 2 errors in 1 file (checked 1396 source files)
```

--- a/pandas/io/formats/printing.py
+++ b/pandas/io/formats/printing.py
@@ -263,7 +263,9 @@ def enable_data_resource_formatter(enable: bool) -> None:
 
             class TableSchemaFormatter(BaseFormatter):
                 print_method = ObjectName("_repr_data_resource_")
-                _return_type = (dict,)
+                # expression has type "Tuple[Type[Dict[Any, Any]], Type[str]]",
+                # base class "BaseFormatter" defined the type as "Type[str]"
+                _return_type = (dict,)  # type: ignore[assignment]
 
             # register it:
             formatters[mimetype] = TableSchemaFormatter()

--- a/pandas/io/formats/printing.py
+++ b/pandas/io/formats/printing.py
@@ -263,9 +263,7 @@ def enable_data_resource_formatter(enable: bool) -> None:
 
             class TableSchemaFormatter(BaseFormatter):
                 print_method = ObjectName("_repr_data_resource_")
-                # expression has type "Tuple[Type[Dict[Any, Any]], Type[str]]",
-                # base class "BaseFormatter" defined the type as "Type[str]"
-                _return_type = (dict,)  # type: ignore[assignment]
+                _return_type = (dict,)
 
             # register it:
             formatters[mimetype] = TableSchemaFormatter()

--- a/pandas/io/formats/printing.py
+++ b/pandas/io/formats/printing.py
@@ -259,9 +259,10 @@ def enable_data_resource_formatter(enable: bool) -> None:
         if mimetype not in formatters:
             # define tableschema formatter
             from IPython.core.formatters import BaseFormatter
+            from traitlets import ObjectName
 
             class TableSchemaFormatter(BaseFormatter):
-                print_method = "_repr_data_resource_"
+                print_method = ObjectName("_repr_data_resource_")
                 _return_type = (dict,)
 
             # register it:

